org/postgresql/jdbc2/AbstractJdbc2Statement.java

```diff
@@ -3350,7 +3353,20 @@ public void setTime(int i, Time t, java.util.Calendar cal) throws SQLException
         if (cal != null)
             cal = (Calendar)cal.clone();
 
-        bindString(i, connection.getTimestampUtils().toString(cal, t), Oid.UNSPECIFIED);
+        int oid = Oid.UNSPECIFIED;
+
+        // If a PGTime is used, we can define the OID explicitly.
```

Comment:
As per JDBC specification, `setTime(int, Time, Calendar)` is "time with time zone" while `setTime(int, Time)` is just "time".

Surely, we can check for `PGTime` kind of classes, while the default should probably be resolved according to "Calendar/non Calendar" method.